### PR TITLE
Upgrade implications associated with transaction_max_timeout_ms limit update

### PR DESCRIPTION
**Pull request by Feediver1**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2616
Review deadline: July 30

## Page previews
https://deploy-preview-649--redpanda-docs-preview.netlify.app/24.2/upgrade/rolling-upgrade/#review-incompatible-changes

## Checks

- [x] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: modules/upgrade/partials/incompat-changes.adoc [hunk 1]
```diff
@@ -4,6 +4,8 @@
...
+* Starting in version 24.2, xref:manage:cluster-maintenance/manage-throughput.adoc#client-throughput-limits[client throughput limits] are compatible with the https://cwiki.apache.org/confluence/display/KAFKA/KIP-546%3A+Add+Client+Quota+APIs+to+the+Admin+Client[AlterClientQuotas and DescribeClientQuotas^] Kafka APIs. Redpanda determines client throughput limits on a per-broker basis. In earlier versions, client throughput quotas were applied from cluster configurations on a per-shard basis.
+
+* Starting in version 24.2, xref:reference:properties/cluster-properties.adoc#transaction_max_timeout_ms[transaction.max.timeout.ms] defaults to 15 minutes, which limits the timeout a transactional client producer can set when initiating transactions. When performing upgrades to version 24.2, if you have clients with timeouts larger than `transaction_max_timeout_ms`, then new producer init requests will fail due to this limit enforcement. You must either reduce the client set transaction timeout to less than 15 minutes or increase the default `transaction_max_timeout_ms` on the service side.
```
Comment (Feediver1) [inline modules/upgrade/partials/incompat-changes.adoc]:
```suggestion
* Starting in version 24.2, xref:reference:properties/cluster-properties.adoc#transaction_max_timeout_ms[transaction_max_timeout_ms] defaults to 15 minutes, which limits the timeout a transactional client producer can set when initiating transactions. When performing upgrades to version 24.2, if you have clients with timeouts larger than `transaction_max_timeout_ms`, then new producer init requests will fail due to this limit enforcement. You must either reduce the client set transaction timeout to less than 15 minutes or increase the default `transaction_max_timeout_ms` on the service side.
```

File: modules/upgrade/partials/incompat-changes.adoc [hunk 2]
```diff
@@ -4,6 +4,8 @@
...
+* Starting in version 24.2, xref:manage:cluster-maintenance/manage-throughput.adoc#client-throughput-limits[client throughput limits] are compatible with the https://cwiki.apache.org/confluence/display/KAFKA/KIP-546%3A+Add+Client+Quota+APIs+to+the+Admin+Client[AlterClientQuotas and DescribeClientQuotas^] Kafka APIs. Redpanda determines client throughput limits on a per-broker basis. In earlier versions, client throughput quotas were applied from cluster configurations on a per-shard basis.
+
+* Starting in version 24.2, xref:reference:properties/cluster-properties.adoc#transaction_max_timeout_ms[transaction_max_timeout_ms] defaults to 15 minutes, which limits the timeout a transactional client producer can set when initiating transactions. When performing upgrades to version 24.2, if you have clients with timeouts larger than `transaction_max_timeout_ms`, then new producer init requests will fail due to this limit enforcement. You must either reduce the client set transaction timeout to less than 15 minutes or increase the default `transaction_max_timeout_ms` on the service side.
```
Comment (micheleRP) [inline modules/upgrade/partials/incompat-changes.adoc]:
```suggestion
* Starting in version 24.2, xref:reference:properties/cluster-properties.adoc#transaction_max_timeout_ms[transaction_max_timeout_ms] defaults to 15 minutes, which limits the timeout a client producer can set when initiating transactions. When performing upgrades to version 24.2, if you have clients with timeouts larger than `transaction_max_timeout_ms`, then new producer initialization requests will fail due to this limit enforcement. Either reduce the client set transaction timeout to less than 15 minutes or increase the default `transaction_max_timeout_ms` on the service side.
```

**Review by micheleRP — APPROVED** [review]
looks good, little suggestions you can consider

File: modules/upgrade/partials/incompat-changes.adoc (hunk 2 above)
Comment (bharathv) [reply modules/upgrade/partials/incompat-changes.adoc]: > which limits the timeout a client producer can set when initiating transactions

instead can we say "which limits the timeout a transactional producer can set during initialization"

**Review by bharathv — COMMENTED** [review]
lgtm, minor suggestion.

**Pull request merged**